extensions/elasticsearch/elasticsearch-7/src/test/java/com/hazelcast/jet/elastic/AuthElasticSinksTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.elastic;
+
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.JetTestInstanceFactory;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.elastic.CommonElasticSinksTest.TestItem;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sink;
+import com.hazelcast.jet.pipeline.test.TestSources;
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.WriteRequest.RefreshPolicy;
+import org.elasticsearch.client.RestClientBuilder;
+import org.junit.After;
+import org.junit.Test;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import static com.hazelcast.jet.elastic.ElasticClients.client;
+import static com.hazelcast.jet.elastic.ElasticSupport.ELASTICSEARCH_IMAGE;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class AuthElasticSinksTest extends BaseElasticTest {
+
+    /**
+     * Using elastic container configured with security enabled
+     */
+    public static Supplier<ElasticsearchContainer> elastic = Util.memoize(() -> {
+        ElasticsearchContainer elastic = new ElasticsearchContainer(ELASTICSEARCH_IMAGE)
+                .withEnv("ELASTIC_USERNAME", "elastic")
+                .withEnv("ELASTIC_PASSWORD", "SuperSecret")
+                .withEnv("xpack.security.enabled", "true");
+
+        elastic.start();
+        Runtime.getRuntime().addShutdownHook(new Thread(elastic::stop));
```

Comment:
Why don't we create the container at `@BeforeClass` phase and stop it at `@AfterClass` ? 
Currently we wait for all the test suite to complete and then stop the container via shutdown hook.

Reply:
This is to avoid starting the container multiple times for different classes.

There were separate secured instances in `AuthElasticSinksTest` and `AuthElasticSourcesTest`, moved this to `ElasticSupport`.